persistence-api/src/main/java/io/stargate/db/datastore/PersistenceBackedResultSet.java

```diff
@@ -159,7 +161,7 @@ private boolean hasNextRow() {
   private Row nextRow() {
     while (true) {
       Row nextRow = fetchedRows.pollFirst();
-      if (nextRow != null) {
+      if (nextRow != null && (authzFilter == null || authzFilter.test(nextRow))) {
```

Comment:
Is there any particular reason to avoid filtering in `processNewPage`?... that would allow us to have the filtering logic only in one place in this class :thinking:

Reply:
I was going to but `processNewPage` is only called from the constructor which I don't really have a good way to get the authzFilter into.

Reply:
Since we have filtering in two places now, could you add unit tests for this class to make sure all call paths are covered by the filter?

Reply:
Sure

Reply:
Added https://github.com/stargate/stargate/pull/386/files#diff-e692aeca1c123715ea4b6cc35cabafe62af29142e07261996a2412566aca36a7L4